### DISCUSSION TITLE
Getting rtorrent to run in a server-client configuration

### DISCUSSION THREAD

**Discussion by mrjpaxton** [rakshasa/rtorrent#1222] [Q&A]
Hey, I was wondering if I can get rtorrent to work in a client and server configuration specifically on Debian without installing too many extra packages.

Please note that Debian does not package `rtorrentd` separately, but I'm aware of it. I'd have to compile it from source.

So I have a custom NAS, and unfortunately it's running Armbian because I have to run it on there, but it's close enough to Debian! Anyways, I wanted to try to run it in daemon mode by specifying `system.daemon.set = true`, and although there's not a lot of documentation about how to use it, I wanted to try it out at least. If this doesn't help with anything, I'll just leave it disabled.

I also tried setting up a unit file for rtorrent with systemd here - https://wiki.archlinux.org/title/RTorrent#Systemd_service_for_a_headless_server

But it doesn't work right. Just gives me errors like, "rtorrent.service: Failed to determine user credentials: No such process", and " rtorrent.service: Failed at step USER spawning /bin/rm: No such process", and "rtorrent.service: Failed at step USER spawning /usr/bin/rtorrent: No such process". Not sure what all of that means in terms of the actual *problem* exactly. But regardless, it's not working right now.

There's also a `bind` option. Now I'm not sure what this does exactly, and I'm not a big networking person, but I was hoping that somehow I could achieve what I want using this.

So does `bind` have to be used "server-side", so that it binds to localhost, and then I can use the `ip` option for whatever socket thing rtorrent uses (if it uses a socket), or does `bind` have to be used "client-side" to connect to rtorrent running on my NAS?

I am also aware of containers and Docker stuff, while I'm not completely opposed to trying it out, I would prefer something more lightweight or linux DIY personally, without it being a completely insecure hack, because I just want it to work reliably, and I'm getting tired of having to run rtorrent individually as a standalone Torrent client on all of my machines, I figured it would be nice to have it more centralized.

**Comment by beadon** [comment]
There are many ways to run rtorrent as a daemon.

Once of these ways is to wrap it in the utility, "screen", and create a systemd compatible file




file:   /etc/systemd/system
user created:  rtorrent
group created: fileadmin
- user is a part of this group
- target filesystem has multiple users allowing read / write access

```

[Unit]
Description=rTorrent
After=network.target

[Service]
User=rtorrent
Type=forking
KillMode=none
ExecStart=/usr/bin/screen -d -m -fa -S rtorrent /usr/bin/rtorrent
ExecStop=/usr/bin/killall -w -s 2 /usr/bin/rtorrent
WorkingDirectory=/home/rtorrent

[Install]
WantedBy=default.target

```
